src/buffer.rs

```diff
@@ -315,3 +334,75 @@ pub enum EncMacroblockParameterBuffer {
     /// Abstraction over `VAEncMacroblockParameterBufferH264`
     H264(h264::EncMacroblockParameterBufferH264),
 }
+
+/// Helper to access a single segment of mapped coded buffer
+pub struct CodedBufferSegment<'s> {
+    pub bit_offset: u32,
+    pub status: u32,
+    pub buf: &'s [u8],
+}
+
+/// Helper to access segments of mapped coded buffer
+pub struct CodedBuffer<'p, 's: 'p, D: SurfaceMemoryDescriptor, T: Borrow<Surface<D>>> {
```

Comment:
I think you can use the lifetime `'p` for `segments` as well and remove `'s` entirely.

Reply:
Done!